File: Challenges/Problem_16/Power_digit_sum.js
// KuntaNgcobo
// Power_digit_sum

const powerDigitSum = exponent => {
  let number = 2n;
  let summer = 0;
  for(var i = 1; i < exponent; i++){
    number += number;
  }

  /* Should be, but the firebox console will not allow for that
    let number = BigInt(2)**BigInt(exponent)
  */

  for(let unit of number + ''){
    summer += parseInt(unit);
  } 
  //console.log(summer);
  return summer;
}

/*
console.log(powerDigitSum(15));
console.log(powerDigitSum(128));
console.log(powerDigitSum(1000));
*/

File: Challenges/Problem_16/Power_digit_sum.py
# KuntaNgcobo
# Power_digit_sum

def powerDigitSum(exponent):
	prod = str(2**exponent)
	sums = 0

	for unit in prod:
		sums += int(unit)

	return sums

'''
print(powerDigitSum(15));
print(powerDigitSum(128));
print(powerDigitSum(1000));
'''